### PR TITLE
Refactors belt harness to attach anything

### PR DESCRIPTION
**Pull request by QualityVan**
## About The Pull Request
Rather than just functioning the exact same as a magharn, you can now attach any item to a belt harness and it'll try to be equipped to suit storage and back when dropped. Don't try throwing things after you attached them though.
There's a big red message warning you to attach something when you equip the belt. Will this stop people asking why their belt didn't work? Probably not.
Moves mob pronoun helper procs up to atom, since that's where gender exists
Fixes storage being improperly unequipped on clickdrag, which mostly mattered for signals.

## Why It's Good For The Game
Slightly more multipurpose equipment, causes PC mains to erupt into tears of joy. You're still spending your belt and back slot for it though.
You could use it for melee weapons that aren't vali too I guess. Won't help for anything that doesn't go in suit storage or back slot of course.

## Changelog

:cl:
refactor: Belt harnesses need an item to be attached to it manually, but any item works - so long as it fits in suit storage or back slot.
/:cl:

### CODE REVIEW

**Comment by novaepee** [comment]
Plasma cutter, here we go!

Would be funny if you try to throw nade, only for belt harness to bring nade back to you.

**Comment by FIXTHERADBOXPLEASE** [comment]
Incredible powercreep for plasmacutters. There are MANY marine tools that can only be pushed by aciding or dragging them away, merging this is denying xenos the chance to affect the game in their favor.

**Comment by Aquilar** [comment]
this is too much of a buff to synths.

**Comment by Tailiik** [comment]
Melting a plasma cutter is one of the best equipment denials xenos can do, engineer main powercreep pr.

**Comment by Guidesu** [comment]
> Melting a plasma cutter is one of the best equipment denials xenos can do, engineer main powercreep pr.

I JUST want it to my machete

**Comment by FIXTHERADBOXPLEASE** [comment]
> magharn PC that isn't even a nodrop like vali machete, powerfist, or pistol lace

There's a very good reason for this. I recommend maintainers ignore the engineer main that spends all his points on 3+ plasmacutters and suicides upon losing any.

**Comment by Strayed-Raven** [comment]
You have to completely ditch a belt slot for a niche build. Sure why not?

**Comment by Losenis** [comment]
>I recommend maintainers ignore the engineer main that spends all his points on 3+ plasmacutters and suicides upon losing any.

You keep accusing me of things with a loadout I don't even run let alone suicide on a support role instead of going shipside to cryo. This is the second time already. Are you really sure you're not confusing me for someone else?

>Anyone using a plasmacutter specs exclusively into using it

It's 20 points from 75 you get. I shove one into the dispenser and actually forget about it with the leftover points I have from tele recharger hicap goggles and other shit.

>(like himself buying 3+,)

Bruh what the fuck is this fixation, are you okay

**Comment by FIXTHERADBOXPLEASE** [comment]
> You have to completely ditch a belt slot for a niche build. Sure why not?

... In exchange for making one of the strongest marine power items completely impossible for the entire other half of the game to punish.
Ignoring the fact that TGMC is a marines vs Xenos game, and that there's an entire other team playing the opposing force that requires the ability to counter the marines, is not a good mentality for the server.

**Comment by FIXTHERADBOXPLEASE** [comment]
> Bruh what the fuck is this fixation, are you okay
I regularly the suicided, DNR body of Tanya 'Urist' after painstakingly dragging away all his plasma cutters one by one as Wraith. Unless I'm wrong, that's your static.

**Comment by Losenis** [comment]
>I regularly the suicided, DNR body of Tanya 'Urist' after painstakingly dragging away all his plasma cutters one by one as Wraith.

No, you don't. I now know for a fact you're deliberatedly making shit up, because the one time I went 3+ plasma cutters was a round I got RO to buy me enough to fill up a dispenser and didn't lose any of them and survived the round to a marine major.

**Comment by FIXTHERADBOXPLEASE** [comment]
> No, you don't. I now know for a fact you're deliberatedly making shit up, because the one time I went 3+ plasma cutters was a round I got RO to buy me enough to fill up a dispenser and didn't lose any of them and survived the round to a marine major.

You regularly boast about spending all your engineering points on plasma cutters on the discord, and I have to outplay you multiple times to get rid of multiple plasma cutters in-game. What more do I have to say?

**Comment by usnpeepoo** [comment]
This is storage power creep, and also denies the ability for xenos to take out special tool users because they somehow have a mag harness on it. -1

**Comment by Losenis** [comment]
The belt slot item that has no storage is storage powercreep?

**Comment by MathiasOrtega** [comment]
This makes primo rouny great again. Go get ‘em loot goblin.

**Comment by Autisem** [comment]
Trading in a belt on the engineer who already has issues storing things not inside there dispenser/backpack is a worthwhile trade imo. this does not deny being able to disable the tool if you just kill the engineer. this just prevents being a rouny poucing an engi and running off with it. compared to other things this is not alot of powercreep

File: code/game/objects/items/marine_gear.dm [hunk 1]
```diff
@@ -245,17 +245,112 @@
...
+		span_warning("[attached_item] smack[attached_item.p_s()] you in the face as [src] pulls it back to you."))
+	if(!ishuman(user))
+		return
+	var/mob/living/carbon/human/huser = user
+	huser.Knockdown(15)
 
 /obj/item/belt_harness/marine
 	name = "\improper M45 pattern belt harness"
-	desc = "A shoulder worn strap with clamps that can attach to a gun. Should keep you from losing your weapon, hopefully."
+	desc = "A shoulder worn strap with clamps that can attach to most anything. Should keep you from losing your weapon, hopefully."
```
Comment (Autisem) [inline code/game/objects/items/marine_gear.dm:345]: should it not be "almost"?

**Comment by Coastfront** [comment]
And a billion xeno mains convulsed, spasming with pain uncontrollable, for one of their greatest ails has grown yet further.

The plasmacutter is a very powerful tool, in which the user holds the power to make xeno mazes irrelevent (ignore flamethrowers for a minute, thats a different arguement all together.). Assuming an engineer convinces even 1 or 2 marine buddies to stick near him, he can RELATIVELY CONSISTANTLY stay alive, get pounced by a woyer/rouny: Gang backs you up and pulls you back/chases the woyer, unless there ar emore xenos backing woyer up:  engie gets away.

Thus, plasmacutter further has to deal with the challenge of xenos stunning the engie, and just fucking gunning it away with his pc.  Likely to be met with pursuit, however it is a signifigantly simpler and easier task to get away with over putting down a marine while under-fire from several others.

With this, the dynamic changes SOLEY to "the engineer must fucking die", and if the engineer has their bodyguard detail, it is moderately harder to accomplish "killing an entire marine and stopping them from getting revived" over "take a single item from its wielder."
On the arguement of "But its sacrificing so much just for the engie to use a pc!" There are (usually) multiple engineers to fufill the array of all of engie's duties.
If an engie goes plasmacutter engie, they are intentionally, and partially neglecting their other duties in exchange to have an EXTREMELY POWERFUL cover-clearing device. They are making up for their absence in their other roles by being considerably more effective at a specific type of combat role.

An undisarmable pc engie means that xenos HAVE TO priority target putting them down, before they quickly run out of walls to hide behind, and if that engie has teammates guarding them its a fuckin pain in the ass to kill them without having to risk a beno dying OR  having to get a squad of xenos to gank them, let alone that they can be defibbed if they're intact enough and are found fast enough.

Having belt attach to LITERALLY ALMOST ANY OTHER ITEM but plasmacutters, wouldn't be as massive an upset to things as funny undisarmable plasmacutter engineer

if i accidentally said "no drop" at any point in this rant im aware this isnt no-drop its fucking 12 am and im tired.

**Comment by Losenis** [comment]
>There are (usually) multiple engineers to fufill the array of all of engie's duties.

And they can't tell they have to actually mend the wires before closing the APC

**Comment by usnpeepoo** [comment]
You're not sacrificing shit bruh it's just a fucking belt, there are pouches for everything and normal marines can already play with the mag harness belt without worrying about space. And even with that, the engineer could just idk grab another belt that they'd need? Or other items to carry their shit? It's not rocket science that it'll be used to solely never lose their plasma cutter and they'll make up for it by just finding more ways to carry what they do normally. This is power creep at its finest and completely prevents xenos from being able to prevent map wall destroying items from being used because it will quite literally never be dropped. It shouldn't be used with any form of mag harness at all

**Comment by Losenis** [comment]
Have you tried killing the engi

**Review by Lumipharon — CHANGES_REQUESTED** [review]
Looking good, sorry for the wait, I thought Tivi was doing something with it.

**Review by Lumipharon — APPROVED** [review]

**Pull request merged**